### DISCUSSION TITLE
Adding only one layer

### DISCUSSION THREAD

**Discussion by RamaKrishnaBodavula** [peterLaurence/MapCompose#90] [Q&A] [answered]
I have only one layer / map image (png) to add, no seperated row and col images. Can we add it at the center instead of top left position. Or is there any other way for handling single layer map images.

Thank you.

**Comment by p-lr** [comment, marked as answer]
Although this library is almost usable for this usage, the fact that only square tiles are supported makes it not appropriate for now.
However, if your single image has equal width and height, setting the tile size to the image size, and scrolling to the center should work.
In the example below, the single image is a square of 1920px width.

```kotlin
val state: MapState by mutableStateOf(
    MapState(1, 1920, 1920, tileSize = 1920) {
        scale(1f)
        scroll(0.5, 0.5)
    }.apply {
        addLayer(tileStreamProvider)
    }
)
```

**Reply by p-lr** [reply]
Instead, I would recommend to [split your single image into tiles](https://github.com/p-lr/MapView/blob/master/doc/libvips.md).

**Reply by RamaKrishnaBodavula** [reply]
The image is not exactly a square / has equal width and height. And splitting the image is not something I can do due to my requirement. But thank you for the support.

**Reply by RamaKrishnaBodavula** [reply]
Is there any other library that you know of or recommend that fits this usage case.

**Reply by Nohus** [reply]
It is unclear what you want to achieve, or why you can't split the image into tiles. If you explain your requirements, you might get better answers.
